tritonbench/operators/gemm/tlx_matmul.py

```diff
@@ -174,42 +187,45 @@ def _process_tile_epilogue_inner(
 ):
     """Process epilogue for a single tile."""
     pid_m, pid_n = _compute_pid(tile_id, num_pid_in_group, num_pid_m, GROUP_SIZE_M)
-    offs_am = pid_m * BLOCK_SIZE_M
     offs_bn = pid_n * BLOCK_SIZE_N
+    BLOCK_M_SPLIT: tl.constexpr = BLOCK_SIZE_M // NUM_MMA_GROUPS
 
-    tlx.barrier_wait(tmem_full_bars[cur_tmem_buf], tmem_read_phase)
-
-    # load the result from TMEM to registers
-    acc_tmem = tmem_buffers[cur_tmem_buf]
     slice_size: tl.constexpr = BLOCK_SIZE_N // EPILOGUE_SUBTILE
-    for slice_id in tl.static_range(EPILOGUE_SUBTILE):
-        acc_tmem_subslice = tlx.local_slice(
-            acc_tmem,
-            [0, slice_id * slice_size],
-            [BLOCK_SIZE_M, slice_size],
-        )
-        result = tlx.local_load(acc_tmem_subslice)
-        c = result.to(tl.float16)
-        c_desc.store([offs_am, offs_bn + slice_id * slice_size], c)
+    for group_id in tl.static_range(NUM_MMA_GROUPS):
+        # Wait for TMEM to be filled
+        buf_idx = group_id * NUM_TMEM_BUFFERS + cur_tmem_buf
+        tlx.barrier_wait(tmem_full_bars[buf_idx], tmem_read_phase)
+
+        # load the result from TMEM to registers
+        acc_tmem = tmem_buffers[buf_idx]
+        offs_am = pid_m * BLOCK_SIZE_M + group_id * BLOCK_M_SPLIT
+        for slice_id in tl.static_range(EPILOGUE_SUBTILE):
+            acc_tmem_subslice = tlx.local_slice(
+                acc_tmem,
+                [0, slice_id * slice_size],
+                [BLOCK_M_SPLIT, slice_size],
+            )
+            result = tlx.local_load(acc_tmem_subslice)
+            c = result.to(tl.float16)
```

Comment:
Nit: Use `c_desc` for the dtype instead? Seems like this kernel should support bf16 automatically as well.